File: core/src/com/broll/gainea/server/core/cards/events/E_SpawnGoddrake.java
package com.broll.gainea.server.core.cards.events;

import com.broll.gainea.server.core.GameContainer;
import com.broll.gainea.server.core.cards.EventCard;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.objects.GodDragon;
import com.broll.gainea.server.core.utils.LocationUtils;
import com.broll.gainea.server.core.utils.UnitControl;

public class E_SpawnGoddrake extends EventCard {
    public E_SpawnGoddrake() {
        super(59, "Gaineas Herrscher", "Der Götterdrache erscheint!");
    }

    @Override
    protected void play() {
        GodDragon dragon = new GodDragon();
        Location location = LocationUtils.getRandomFree(game.getMap().getAllAreas());
        UnitControl.spawn(game, dragon, location);
    }

    public static boolean isGoddrakeAlive(GameContainer game) {
        return game.getObjects().stream().filter(it -> it instanceof GodDragon).findAny().isPresent();
    }
}

File: server/src/main/com/broll/gainea/server/core/cards/events/E_SpawnGoddrake.kt
package com.broll.gainea.server.core.cards.events

import com.broll.gainea.server.core.Game
import com.broll.gainea.server.core.cards.EventCard
import com.broll.gainea.server.core.objects.monster.GodDragon
import com.broll.gainea.server.core.utils.UnitControl.spawn
import com.broll.gainea.server.core.utils.getRandomFree

class E_SpawnGoddrake : EventCard(59, "Gaineas Herrscher", "Der Götterdrache erscheint!") {
    override fun play() {
        val dragon = GodDragon(game.neutralPlayer)
        game.map.allAreas.getRandomFree()?.let {
            game.spawn(dragon, it) { nt -> nt.sound = "goddrake.ogg" }
        }
    }

    companion object {
        fun isGoddrakeAlive(game: Game) = game.objects.any { it is GodDragon }

    }
}
